controllers/articles-controller.js

```diff
@@ -25,12 +25,31 @@ exports.getArticles = (req, res, next) => {
 exports.getComments = (req, res, next) => {
     const id = req.params.article_id
     const commentsPromises = [selectArticlesComments(id), checkExists("articles", "article_id", id)]
-    Promise.all(commentsPromises).then((resolvedPromises) => {
-        console.log(resolvedPromises)
+    Promise.all(commentsPromises)
+    .then((resolvedPromises) => {
         const comments = resolvedPromises[0]
         res.status(200).send({ comments })
     })
     .catch((err) => {
         next(err)
     })
+}
+
+exports.postComment = (req, res, next) => {
+    const body = req.body.body
+    const author = req.body.username
+    const article_id = req.params.article_id
+    const commentPromise = addCommentByArticleId(body, author, article_id)
+    const articleExistencePromise = checkExists("articles", "article_id", article_id);
+    const userExistencePromise = checkExists("users", "username", author)
+    const commentsPromises = [commentPromise, articleExistencePromise, userExistencePromise]
+    Promise.all(commentsPromises)
+    .then((resolvedPromises) => {
+        const commentArray = resolvedPromises[0]
+        const comment = commentArray[0]
+        res.status(201).send({ comment })
+    })
```

Comment:
This logic belongs in your model - the only thing the controller should do is invoke the model function with the right arguments, the logic for checking the article and user exists should all be part of the model.